### PR TITLE
add scale to EDA's display labels

### PR DESCRIPTION
**Pull request by moontrip**
This will address https://github.com/VEuPathDB/web-monorepo/issues/450.

It was hard to find where the scale info (metadata) is used but at least I could manage to find one study, Mal-ED study. After checking the codes carefully, I realized that an easy solution was to change `variableDisplayWithUnit()` so that all (or most) Vizs show scale info at various display labels such as axis title, legend title, variable coverage table etc. Below are screenshot with Scatterplot for before and after this PR.

I saw that there were cases where units are empty while scale exists. In that case, I just used scale with parenthesis: e.g., Polio serotype... in the "After" screenshot. 

### **- Before**
MAL-ED scale01

### **- After**
MAL-ED scale01-PR

### CODE REVIEW

File: packages/libs/eda/src/lib/core/utils/variable-display.ts [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+        return variable.displayName + ' (' + variable.units + ')';
+      }
+    } else {
+      if (variable.scale && variable.scale != null) {
+        return variable.displayName + ' (' + variable.scale + ')';
+      } else {
+        return variable?.displayName;
+      }
+    }
+  } else return variable?.displayName;
```
Comment (dmfalke) [inline packages/libs/eda/src/lib/core/utils/variable-display.ts:28]: There is quite a bit of nesting of if-statements here. I think we can reduce the this significantly

```suggestion
  if (variable == null) return;
  if (!NumberVariable.is(variable)) return variable.displayName;
  
  const parentheticalParts = [ variable.units, variable.scale ].filter(Boolean);
  
  if (parentheticalParts.length) return `${variable.displayName} (${parentheticalParts.join(', ')})`;

  return variable.displayName;
```

**Review by dmfalke — CHANGES_REQUESTED** [review]
Overall, this looks good. I added a suggested change, to simplify the code. Let me know what you think. Thanks!

**Comment by moontrip** [comment]
> Overall, this looks good. I added a suggested change, to simplify the code. Let me know what you think. Thanks!

Hi again @dmfalke 👋 Thank you for your review and suggestion! I saw the technique, array.filter(Boolean), somewhere before and I think it would work quite well in this case. Will check it out next week 👍

**Comment by moontrip** [comment]
@dmfalke I have addressed your feedbacks. It would be great if you can check it out 👍

**Review by dmfalke — APPROVED** [review]

**Comment by moontrip** [comment]
Thank you for your review @dmfalke 👍

**Pull request merged**